loky/backend/context.py

```diff
@@ -146,11 +160,81 @@ def cpu_count():
             # float in python2.7. (See issue #165)
             cpu_count_cfs = int(math.ceil(cfs_quota_us / cfs_period_us))
 
-    # User defined soft-limit passed as an loky specific environment variable.
+    # User defined soft-limit passed as a loky specific environment variable.
     cpu_count_loky = int(os.environ.get('LOKY_MAX_CPU_COUNT', cpu_count_mp))
-    aggregate_cpu_count = min(cpu_count_mp, cpu_count_affinity, cpu_count_cfs,
-                              cpu_count_loky)
-    return max(aggregate_cpu_count, 1)
+
+    cpu_count_user = min(cpu_count_affinity, cpu_count_cfs, cpu_count_loky)
+    aggregate_cpu_count = min(cpu_count_mp, cpu_count_user)
+
+    if only_physical_cores:
+        cpu_count_physical, error_message = _count_physical_cores()
+        if cpu_count_user < cpu_count_mp:
+            # Respect user setting
+            cpu_count = max(cpu_count_user, 1)
+        elif cpu_count_physical == "not found":
+            # Fallback to default behavior
+            warnings.warn("Could not find the number of physical cores. "
+                          "Returning the number of logical cores instead.\n"
+                          + error_message)
```

Comment:
I am worried this will be too verbose for programs that call `loky.cpu_count()` many times.

I think it would be better to only warn the first time we get the error. I think it would be enough to not cache the error message and to not warn if `error_message` is left to None.